client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +47,125 @@ export default class UCIMap extends PureComponent {
         selected_acronym: '',
         eventsInCalendar: AppStore.getEventsInCalendar(),
         currentScheduleIndex: AppStore.getCurrentScheduleIndex(),
+        poly: [],
+    };
+
+    getRoute = (day) => {
+        let index = 0;
+        let coords = '';
+        let coords_array = [];
+        let colors = [];
+        this.state.eventsInCalendar
+            .sort((event, event2) => event.start - event2.start)
+            .forEach((event) => {
+                // Filter out those in a different schedule or those not on a certain day (mon, tue, etc)
+                if (
+                    event.isCustomEvent ||
+                    !event.scheduleIndices.includes(this.state.currentScheduleIndex) ||
+                    !event.start.toString().includes(DAYS[day])
+                )
+                    return;
+
+                // Get building code, get id of building code, which will get us the building data from buildingCatalogue
+                const buildingCode = event.bldg.split(' ')[0];
+                const id = locations[buildingCode];
+                const locationData = buildingCatalogue[id];
+
+                if (locationData === undefined) return;
+
+                colors.push(event.color);
+
+                if (day) {
+                    if (coords) coords += ';';
+                    coords += locationData.lng + ',' + locationData.lat;
+                    coords_array.push([locationData.lat, locationData.lng]);
+                }
+                index++;
+            });
+        if (day && index > 1) {
+            var url = new URL(DIRECTIONS_ENDPOINT + encodeURIComponent(coords));
+
+            url.search = new URLSearchParams({
+                alternatives: false,
+                geometries: 'geojson',
+                steps: false,
+                access_token: ACCESS_TOKEN,
+            }).toString();
+
+            fetch(url, {
+                method: 'GET',
+                headers: { 'Content-Type': 'application/json' },
+            }).then((response) => {
+                response.json().then((obj) => {
+                    let latlng = obj['routes'][0]['geometry']['coordinates'];
+                    let waypoint = 0;
+                    let path = [
+                        [[obj['waypoints'][waypoint]['location'][1], obj['waypoints'][waypoint]['location'][0]]],
+                    ];
+                    let poly = [];
+                    poly.push(<Polyline color={colors[0]} positions={[path[0][0], coords_array[0]]} dashArray="4" />);
+
+                    for (let i of latlng) {
```

Comment:
Hard to understand what `i` represents at first glance. We can actually destructure here to pull out the lat and lng. The rest of the code in this for loop should be updated accordingly to use lat/lng instead of i[0/1]

```suggestion
                    for (let [lat, lng] of coordinates) {
```
_Notes: this assumes `latlng` is renamed to `coordinates`, per my previous comment_